cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -207,12 +234,15 @@ private AuthenticationClient getAuthenticationClient(ConnectionConfig connection
     return authenticationClient;
   }
 
-  private AccessToken getSavedAccessToken(String hostname) {
+  private AccessTokenWrapper getSavedAccessToken(String hostname) {
     File file = getAccessTokenFile(hostname);
     if (file.exists() && file.canRead()) {
       try {
-        String tokenString = Joiner.on("").join(Files.readLines(file, Charsets.UTF_8));
-        return new AccessToken(tokenString, -1L, null);
+        String fileString = Joiner.on("").join(Files.readLines(file, Charsets.UTF_8));
+        int splitIndex = fileString.indexOf("username:");
```

Comment:
Looks like the existing code is bad. The logic for writing and reading of AccessToken is different (stripping newlines on read). In fact, on write we should just serialize the `AccessToken` (or `UserAccessToken` in the new code) using `Gson` and deserialize it back on read.